internal/storage/sql/mysql/mysql.go

```diff
@@ -172,7 +172,7 @@ func (s *Store) CreateDistribution(ctx context.Context, r *flipt.CreateDistribut
 		var merr *mysql.MySQLError
 
 		if errors.As(err, &merr) && merr.Number == constraintForeignKeyErrCode {
-			return nil, errs.ErrNotFoundf("rule %q", r.RuleId)
+			return nil, errs.ErrNotFoundf("variant %q, rule %q, flag %q/%q in namespace", r.VariantId, r.RuleId, r.NamespaceKey, r.FlagKey)
```

Comment:
do these db specific error handlers need to change as well? I would think the `distributionValidationHelper` called before here would return the correct error to the caller?

Reply:
makes sense, could we make these match the newly updated error messages?

```
errs.ErrNotFoundf("variant %q, rule %q, flag %q in namespace %q", r.VariantId, r.RuleId, r.FlagKey, r.NamespaceKey)
```

Reply:
@markphelps Sounds great!